exporter/awsexporter/aws_xray.go

```diff
@@ -146,13 +146,13 @@ func transformConfigToXRayOptions(axrCfg *awsXRayConfig) (xopts []xray.Option, e
 
 // ExportSpans is the method that translates OpenCensus-Proto Traces into AWS X-Ray spans.
 // It uniquely maintains
-func (axe *awsXRayExporter) ExportSpans(ctx context.Context, td data.TraceData) (xerr error) {
+func (axe *awsXRayExporter) ProcessTraceData(ctx context.Context, td data.TraceData) (xerr error) {
 	ctx, span := trace.StartSpan(ctx,
 		"opencensus.service.exporter.aws_xray.ExportSpans",
 		trace.WithSampler(trace.NeverSample()))
 
 	defer func() {
-		if xerr != nil {
+		if xerr != nil && span.IsRecordingEvents() {
```

Comment:
q: I assume that this check doesn't affect the zpages, correct?

Reply:
It doesn't, even for zpages the Span should record the events in order to be able to display them.